Spring security

juandbc opened the pull request:
## Summary
- Add spring security dependency
- Add password attribute to User model
- Implemented custom security configuration to retrieve user credentials from the database
- Refactor test to avoid duplication
- Add tests for controllers

## Trello Card
https://trello.com/c/CzY85bvD/23-integrar-spring-security

Commit:
Add spring security dependency
Add custom web security
Refactor test to avoid duplication
Add tests for controllers
Add password attribute to User model

jeymarsonwolox commented on src/main/java/wolox/training/models/User.java:
```diff
@@ -64,6 +65,10 @@
 
+    @Getter
+    @Setter
+    private String password;
```
add `@JsonProperty(access = Access.WRITE_ONLY)`, to not show the password when the user is consulted

juandbc replied:
I'll apply it, thanks.

Commit:
Update user model to encode password
Add JSON annotation to not show password in JSON

Gianni-Nicolas commented on src/main/java/wolox/training/models/User.java:
```diff
@@ -64,6 +65,10 @@
 
+    @Getter
+    @Setter
+    private String password;
```
Add a PUT endpoint into UserController that allows updating only the password.

The password must always be modified using a resource other than updating a User

Gianni-Nicolas reviewed (changes requested):
Very good implementation, I just leave a comment 😄

Commit:
Add PUT resource to update user password
Fix credentials validation for HTTP Basic Auth
Add swagger docs for password attribute
